end-to-end-tests/src/instance_launch.rs

```diff
@@ -0,0 +1,258 @@
+                    .send()
+                    .await?
+                    .data,
+            )
+            .into_owned();
+            if data.contains("localshark login:") {
+                Ok(data)
+            } else {
+                Err(Error::NotYet)
+            }
+        },
+        &Duration::from_secs(5),
+        &Duration::from_secs(300),
+    )
+    .await?;
+
+    let host_key = serial
+        .split_once("-----BEGIN SSH HOST KEY KEYS-----")
+        .and_then(|(_, s)| s.split_once("-----END SSH HOST KEY KEYS-----"))
+        .and_then(|(lines, _)| {
+            lines.trim().lines().find(|line| line.starts_with("ssh-ed25519"))
+        })
+        .and_then(|line| line.split_whitespace().nth(1))
+        .context("failed to get SSH host key from serial console")?;
+    eprintln!("host key: ssh-ed25519 {}", host_key);
+    let host_key =
+        PublicKey::parse(b"ssh-ed25519", &base64::decode(host_key)?)?;
+
+    eprintln!("connecting ssh");
+    let mut session = thrussh::client::connect(
+        Default::default(),
+        (ip_addr, 22),
+        SshClient { host_key },
+    )
+    .await?;
+    eprintln!("authenticating ssh");
+    ensure!(
+        session.authenticate_publickey("debian", key).await?,
+        "authentication failed"
+    );
+
+    eprintln!("open session");
+    let mut channel = session.channel_open_session().await?;
+    eprintln!("exec");
+    channel.exec(true, "echo 'Hello, Oxide!' | sudo tee /dev/ttyS0").await?;
+    while let Some(msg) = channel.wait().await {
+        eprintln!("msg: {:?}", msg);
+        match msg {
+            ChannelMsg::Data { data } => {
+                ensure!(
+                    data.as_ref() == b"Hello, Oxide!\n",
+                    "wrong output: {:?}",
+                    data
+                );
+            }
+            ChannelMsg::ExitStatus { exit_status } => {
+                ensure!(exit_status == 0, "exit status {}", exit_status);
+                break;
+            }
+            _ => {}
+        }
+    }
+
+    // sign off
+    eprintln!("disconnecting ssh");
+    channel.eof().await?;
+    session.disconnect(Disconnect::ByApplication, "cya", "en").await?;
+
+    // check that we saw it on the console
+    eprintln!("waiting for serial console");
+    sleep(Duration::from_secs(5)).await;
+    let data = String::from_utf8_lossy(
+        &ctx.client
+            .instance_serial_console()
+            .organization_name(ctx.org_name.clone())
+            .project_name(ctx.project_name.clone())
+            .instance_name(instance.name.clone())
+            .most_recent(1024 * 1024)
+            .max_bytes(1024 * 1024)
+            .send()
+            .await?
+            .data,
+    )
+    .into_owned();
+    ensure!(
+        data.contains("Hello, Oxide!"),
+        "string not seen on console\n{}",
+        data
+    );
+
+    // tear-down
+    eprintln!("tear-down");
+    ctx.client
+        .instance_stop()
```

Comment:
We *should* be deleting the instance before the project / org.

We aren't checking for this yet, but we will be in the future